runtime/binding-mqtt-kafka/src/main/java/io/aklivity/zilla/runtime/binding/mqtt/kafka/internal/stream/MqttKafkaSessionFactory.java

```diff
@@ -3415,6 +3423,61 @@ private KafkaSessionStateProxy(
             super(originId, routedId, delegate);
         }
 
+        @Override
+        protected void onKafkaBegin(
+            BeginFW begin)
+        {
+            final long sequence = begin.sequence();
+            final long acknowledge = begin.acknowledge();
+            final int maximum = begin.maximum();
+            final long traceId = begin.traceId();
+            final long authorization = begin.authorization();
+            final long affinity = begin.affinity();
+            final OctetsFW extension = begin.extension();
+
+            assert acknowledge <= sequence;
+            assert sequence >= replySeq;
+            assert acknowledge >= replyAck;
+
+            replySeq = sequence;
+            replyAck = acknowledge;
+            replyMax = maximum;
+            state = MqttKafkaState.openingReply(state);
+
+            assert replyAck <= replySeq;
+
+            final KafkaBeginExFW kafkaBeginEx = extension.get(kafkaBeginExRO::tryWrap);
+
+            onKafkaBegin:
+            {
+                if (kafkaBeginEx != null)
+                {
+                    assert kafkaBeginEx.kind() == KafkaBeginExFW.KIND_MERGED;
+                    final KafkaMergedBeginExFW kafkaMergedBeginEx = kafkaBeginEx.merged();
+
+                    final KafkaConfigFW cleanupPolicyConfig =
+                        kafkaMergedBeginEx.configs().matchFirst(c -> c.name().equals(CONFIG_NAME_CLEANUP_POLICY));
+                    final String16FW cleanupPolicy = cleanupPolicyConfig != null ? cleanupPolicyConfig.value() : null;
+
+                    if (cleanupPolicy == null || !cleanupPolicy.equals(COMPACT_CLEANUP_POLICY))
+                    {
+                        Flyweight mqttResetEx = mqttSessionResetExRW.wrap(sessionExtBuffer, 0, sessionExtBuffer.capacity())
+                            .typeId(mqttTypeId)
+                            .reasonCode(MQTT_IMPLEMENTATION_SPECIFIC_ERROR)
+                            .reason(MQTT_NON_COMPACT_SESSIONS_TOPIC)
```

Comment:
This reason seems out of scope for an MQTT client, as it is Kafka-specific.

This is more like what we need in the event log where we can describe what went wrong so that the devops can fix it, not the mqtt client.